pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -147,20 +148,31 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
 		ReadOnly:  true,
 	}
 	userContainer.VolumeMounts = append(userContainer.VolumeMounts, sharedVolumeReadMount)
-
 	// Change the CustomSpecStorageUri env variable value to the default model path if present
 	for index, envVar := range userContainer.Env {
 		if envVar.Name == constants.CustomSpecStorageUriEnvVarKey && envVar.Value != "" {
 			userContainer.Env[index].Value = constants.DefaultModelLocalMountPath
 		}
 	}
+	podNamespace := pod.Namespace
+	for _, container := range pod.Spec.Containers {
+		for _, envVar := range container.Env {
+			// Somehow pod namespace is empty when coming into pod mutator, here we need to use
```

Comment:
Do you think this bug might exist in other pod mutators (istio sidecar, etc). I wonder how they've worked around it. Ideally this fix would come through controller-runtime. I wonder if it's a versioning problem